Maven Plugin - the option to set extraClasspath in the plugin configuration isn't working

**Jetty version : 10.0.8**

**Java version/vendor :jdk-11.0.12**

**OS type/version : Windows 10**

**Description**
In version 10.0.8 , the class `org.eclipse.jetty.webapp.WebAppContext` is no more POJO class, 
since there  is an overloading to the setter `setExtraClasspath` :
```
   public void setExtraClasspath(String extraClasspath) throws IOException
    {
        setExtraClasspath(Resource.fromList(extraClasspath, false, this::newResource));
    }

    public void setExtraClasspath(List<Resource> extraClasspath)
    {
        _extraClasspath = extraClasspath;
    }
```
so that , the maven plugin can't set this value , since the sisu-plexus is taking the 1st setter by reflection , assuming there is only one
This bug is not consistent , because if the classloader will find the 1st overload as the 1st setter, it will work , 
but if the 2nd overload will be the 1st setter , so the  sisu-plexus won't be able to construct List<Resource> from the string input
 

**How to reproduce?**
1. Create a jetty maven project,
2. in your pom.xml, add build proflie for jetty for example :
```
        <profile>
            <id>jetty</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.eclipse.jetty</groupId>
                        <artifactId>jetty-maven-plugin</artifactId>
                        <configuration>
                            <webApp>
                                <extraClasspath>/path/to/classpath</extraClasspath>
                            </webApp>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
```
3. run your maven build with this profile, since this bug is not consistent , you may need to try to build several times in order to get the error - 

```
[ERROR] Failed to execute goal org.eclipse.jetty:jetty-maven-plugin:10.0.8:run (default-cli) on project test-jetty: Unable to parse configuration of mojo org.eclipse.jetty:jetty-maven-plugin:10.0.8:run for parameter #: Cannot create instance of class org.eclipse.jetty.util.resource.Resource: InstantiationException -> [Help 1]


```

--- a/jetty-maven-plugin/src/it/jetty-start-mojo-it/jetty-simple-webapp/pom.xml
+++ b/jetty-maven-plugin/src/it/jetty-start-mojo-it/jetty-simple-webapp/pom.xml
@@ -62,6 +62,31 @@
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>copy-random-dependency</id>
+            <phase>generate-resources</phase>
+            <goals>
+              <goal>copy</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <artifactItem>
+                  <groupId>org.junit.jupiter</groupId>
+                  <artifactId>junit-jupiter-engine</artifactId>
+                  <version>@junit.version@</version>
+                  <type>jar</type>
+                  <outputDirectory>${project.build.directory}/dependency</outputDirectory>
+                  <destFileName>junit-jupiter-engine.jar</destFileName>
+                </artifactItem>
+              </artifactItems>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <includes>
@@ -92,6 +117,9 @@
               <goal>start</goal>
             </goals>
             <configuration>
+              <webApp>
+                <extraClasspath>${project.build.directory}/dependency/junit-jupiter-engine.jar</extraClasspath>
+              </webApp>
               <contextXml>${basedir}/src/config/context.xml</contextXml>
               <systemProperties>
                   <jetty.port.file>${jetty.port.file}</jetty.port.file>

--- a/jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/MavenWebAppContext.java
+++ b/jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/MavenWebAppContext.java
@@ -503,4 +503,11 @@ public class MavenWebAppContext extends WebAppContext
             LOG.warn("Problem initializing cdi", e);
         }
     }
+
+    // need to be overridden to avoid Maven reflection issues with super class and override method
+    public void setExtraClasspath(String extraClasspath) throws IOException
+    {
+        super.setExtraClasspath(extraClasspath);
+    }
+
 }

